Failing Azure DevOps Build Test

:clock11: **When did the problem start?**
~2 days ago

<!-- Indicate when the problem started -->

:camera: **Live badge**

<!-- Provide a link to the live badge in plain text and markdown.  -->
https://img.shields.io/azure-devops/build/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/4.svg
![](https://img.shields.io/azure-devops/build/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/4.svg )

:wrench: **Is the live badge working?**
Badge is working, just an unusual color mismatch (passing label but orange badge color)
<!-- Indicate whether or not the live badge is working. -->

:link: **CircleCI link**
First started failing back in https://circleci.com/gh/badges/daily-tests/400
<!-- Provide a link to the failing test in CircleCI. -->

:beetle: **Stack trace**

```
AzureDevOpsBuild partially succeeded build [ GET /totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/4.json ]

AssertionError: color mismatch: expected 'brightgreen' to equal 'orange'
    at Function._expectField (core/service-test-runner/icedfrisby-shields.js:48:51)
    at IcedFrisbyNock.afterJSON.json (core/service-test-runner/icedfrisby-shields.js:41:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:924:10)
    at invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:973:24)
    at Promise (node_modules/icedfrisby/lib/icedfrisby.js:987:7)
    at new Promise (<anonymous>)
    at IcedFrisbyNock._runHooks (node_modules/icedfrisby/lib/icedfrisby.js:946:12)
    at IcedFrisbyNock.run (node_modules/icedfrisby/lib/icedfrisby.js:1238:20)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
```

:bulb: **Possible solution**
~~Not sure yet~~
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->
**Update** I'm not sure why but we're getting a mix of responses from the target Azure DevOps definition that someone set up specifically to cover this "partially succeeded" use case. Some times the response is coming back with "partially succeeded" but other times as "succeeded" resulting in intermittent test failures. 

I don't have access to the target pipeline in Azure DevOps to investigate, but it seems that add an explicit branch to the test badge is resulting in the expected behavior .
<!-- Love Shields? Please consider donating $10 to sustain our activities:
👉  https://opencollective.com/shields -->

--- a/services/azure-devops/azure-devops-build.tester.js
+++ b/services/azure-devops/azure-devops-build.tester.js
@@ -7,14 +7,14 @@ const t = (module.exports = require('../tester').createServiceTester())
 // solely created for Shields.io testing.
 
 t.create('default branch')
-  .get('/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/2.json')
+  .get('/totodem/shields.io/2.json')
   .expectBadge({
     label: 'build',
     message: isBuildStatus,
   })
 
 t.create('named branch')
-  .get('/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/2/master.json')
+  .get('/totodem/shields.io/2/master.json')
   .expectBadge({
     label: 'build',
     message: isBuildStatus,
@@ -34,5 +34,5 @@ t.create('unknown user')
 
 // The following build definition has always a partially succeeded status
 t.create('partially succeeded build')
-  .get('/totodem/8cf3ec0e-d0c2-4fcd-8206-ad204f254a96/4.json')
+  .get('/totodem/shields.io/4/master.json')
   .expectBadge({ label: 'build', message: 'passing', color: 'orange' })

